client/packages/requisitions/src/RequestRequisition/api/api.ts

```diff
@@ -223,14 +223,22 @@ export const getRequestQueries = (sdk: Sdk, storeId: string) => ({
     let result;
     if (draftLine.isCreated) {
       const input = requestParser.toInsertLine(draftLine);
-      result = await sdk.insertRequestLine({
+      const insertResult = await sdk.insertRequestLine({
         storeId,
         input,
       });
 
-      const { insertRequestRequisitionLine } = result || {};
-      if (insertRequestRequisitionLine?.__typename === 'RequisitionLineNode') {
-        return insertRequestRequisitionLine;
+      const { insertRequestRequisitionLine } = insertResult || {};
+      if (insertRequestRequisitionLine?.__typename !== 'RequisitionLineNode') {
+        throw new Error('Failed to create initial requisition line');
+      }
+
+      const updateInput = requestParser.toUpdateLine(draftLine);
+      result = await sdk.updateRequestLine({ storeId, input: updateInput });
+
+      const { updateRequestRequisitionLine } = result || {};
+      if (updateRequestRequisitionLine?.__typename === 'RequisitionLineNode') {
+        return updateRequestRequisitionLine;
       }
```

Comment:
To avoid saving when selecting from autocomplete, we now save only on okButton or save click.
This API handles that by doing an insert, then update to store values. Not sure if ideal, its either this or we write a batch description for it?

Reply:
Oo great point - I actually think this is a good chance to update the API to support creating invoice + lines at same time! I think returns is a good example of this?